upip/upip.py

```diff
@@ -157,6 +161,9 @@ def install_pkg(pkg_spec, install_path):
     f3 = tarfile.TarFile(fileobj=f2)
```

Comment:
Why not reuse local f2 variable here, intsead of making a new f3?  Then there's less Python stack needed, and less risk that there are dangling pointers lef to f2/f3 on the Python value stack.  It also means you don't need to explicitly del f3 and f2, so saves on some bytecode.